test/e2e_node/mount_rro_linux_test.go

```diff
@@ -123,13 +123,9 @@ var _ = SIGDescribe("Mount recursive read-only [LinuxOnly]", framework.WithSeria
 							},
 						},
 					}
-					pod = e2epod.NewPodClient(f).Create(ctx, pod)
-					framework.ExpectNoError(e2epod.WaitForPodContainerToFail(ctx, f.ClientSet, pod.Namespace, pod.Name, 0, "CreateContainerConfigError", framework.PodStartShortTimeout))
-					var err error
-					pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(ctx, pod.Name, metav1.GetOptions{})
-					framework.ExpectNoError(err)
-					gomega.Expect(pod.Status.ContainerStatuses[0].State.Waiting.Message).To(
-						gomega.ContainSubstring("failed to resolve recursive read-only mode: volume \"mnt\" requested recursive read-only mode, but it is not read-only"))
+					_, err := f.ClientSet.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+					gomega.Expect(err.Error()).To(
+						gomega.ContainSubstring("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true"))
```

Comment:
(This fix is distinct from #123848)

Reply:
How about this?
```golang
gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true")))
```

or 

```golang
gomega.Expect(err).To(gomega.MatchError(apierrors.IsForbidden, fmt.Sprintf("the pod %s is forbidden: may only be specified when readOnly is true", pod.Name)))
```

Reply:
or 
```golang
gomega.Expect(err).To(gomega.MatchError("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true"))
```

Reply:
Updated to use `gomega.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("spec.containers[0].volumeMounts.recursiveReadOnly: Forbidden: may only be specified when readOnly is true")))`